packages/react-devtools-scheduling-profiler/src/utils/getBatchRange.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import memoize from 'memoize-one';
+
+import type {BatchUID, Milliseconds, ReactProfilerData} from '../types';
+
+// TODO: Batch duration probably shouldn't include delayed passive effects?
+// It should probably end with the layout effect.
```

Comment:
Actually this comment was from your prototype and we didn't touch it 😆 I've removed it.

Could I clarify what you mean by delayed passive effects? Are these passive effects that are executed some time after the commit is complete?

I'm asking this because this `getBatchRange` implementation (untouched from your prototype) returns a range where the `stopTime` is set to the start time of the next batch. This means that there's a long gap after the last layout and passive effect for that batch.

For example, given this:

![image](https://user-images.githubusercontent.com/12784593/90654339-de2f1c80-e272-11ea-9a1e-b70b8c572a96.png)

For the hovered batch, `getBatchRange`'s `stopTime` will be where the black bar is, not the green bar. Is this correct?

There's a more extreme example at https://github.com/MLH-Fellowship/scheduling-profiler-prototype/issues/124. This screenshot was created by clicking the zoom to batch button on the hovered measures:

![image](https://user-images.githubusercontent.com/12784593/89553912-b9818080-d840-11ea-938f-9a184c71abe4.png)

Reply:
Yeah that's not surprising :smile: I'm just not sure I agree with it anymore. 😆 

> Could I clarify what you mean by delayed passive effects? Are these passive effects that are executed some time after the commit is complete?

Passive effects are run async, next tick/frame, in most cases.

If another render needs to start before passive effects have been flushed, we'll just-in-time flush them before starting on the render work. So I think what you're showing above is that happening.

Although it looks odd, the passive effects *are* part of the work though. And they could be *slow* if you're doing something expensive, so I think it's important for us to include them in the zoomed-in phase.